crates/jetstreamer-source/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+                        num_partitions: rewards.num_partitions.map(|np| yellowstone_grpc_proto::solana::storage::confirmed_block::NumPartitions { num_partitions: np }),
+                    }),
+                    block_time: block_time.map(|bt| UnixTimestamp { timestamp: bt }),
+                    block_height: block_height.map(|bh| BlockHeight { block_height: bh }),
+                    executed_transaction_count,
+                    transactions: vec![],
+                    updated_account_count: 0,
+                    accounts: vec![],
+                    entries: vec![],
+                    entries_count: entry_count,
+                    parent_slot,
+                    parent_blockhash: parent_blockhash.to_string(),
+                }
+            )),
+            created_at: Some(yellowstone_grpc_proto::prost_types::Timestamp::from(std::time::SystemTime::now())),
+        };
+
+                info!(slot, "Sending block update");
+                if let Err(e) = self.tx.send(Ok(update)).await {
+                    let error_msg = format!("Failed to send block update: {}", e);
+                    error!("{}", error_msg);
+                    return Err(Error::ChannelSend(error_msg));
+                }
+            },
+            BlockData::LeaderSkipped { slot } => {
+                debug!(slot, "Skipping leader-skipped slot");
+            },
+        }
+
+        Ok(())
+    }
+
+    async fn process_transaction(&self, tx: TransactionData) -> Result<(), Error> {
+        info!(
+            signature = ?tx.signature,
+            slot = tx.slot,
+            transaction_slot_index = tx.transaction_slot_index,
+            is_vote = tx.is_vote,
+            "Processing transaction"
+        );
+
+        // Check if transaction should be filtered
+        if !self.should_process_transaction(&tx) {
+            debug!(signature = ?tx.signature, slot = tx.slot, "Transaction filtered out");
+            return Ok(());
+        }
+        // TODO: Populate transaction field with protobuf-encoded transaction data
+        let update = SubscribeUpdate {
+            filters: vec![],
+            update_oneof: Some(UpdateOneof::Transaction(
+                yellowstone_grpc_proto::geyser::SubscribeUpdateTransaction {
+                    slot: tx.slot,
+                    transaction: None, //protobuf-encoded tx when available
+                },
+            )),
+            created_at: Some(yellowstone_grpc_proto::prost_types::Timestamp::from(
+                std::time::SystemTime::now(),
+            )),
+        };
+
+        info!(slot = tx.slot, "Sending transaction update");
+        if let Err(e) = self.tx.send(Ok(update)).await {
+            let error_msg = format!("Failed to send transaction update: {}", e);
+            error!("{}", error_msg);
+            return Err(Error::ChannelSend(error_msg));
+        }
+
+        Ok(())
+    }
+
+    fn should_process_transaction(&self, tx: &TransactionData) -> bool {
```

Comment:
Instead of returning boolean this should return Vec<String> to set on the filters key of the subscribe update. If its empty then you can toss out the event and don't send it to the channel.